src/perform_step/symplectic_perform_step.jl

```diff
@@ -153,8 +176,9 @@ end
   @unpack t,dt,f,p = integrator
   @unpack a1,a2,b1,b2 = cache
   duprev,uprev = integrator.uprev.x
+  kuprev = integrator.k[2].x[2]
```

Comment:
these should use fsallast instead of k directly in case the values are updated by the FSAL due to an event. Though it might work because of aliasing.

Reply:
I can change it to `kuprev = integrator.fsllast.x[2]` instead and verify if it still works.
It looked consistent to use the same variable `k.x`, which is filled at the end of all step methods.
I also recognized, that the same lines are coded multiple times to load and store the state and derivatives in most of the algorithms.
Do you think it is worth while to centralize this code in in-lined functions or macros?
Something like `int_state!(integrator...); uprev, duprev, kuprev = load_state(integrator...); store_state!(integrator)` that would allow to handle this `fsal` thing at one spot.

Reply:
As a separate PR, yes. For most algorithms the FSAL handling is a trivial single line: partitioned methods get a little tricker though. But the issue isn't where you placed it, it's about the FSAL updates. If someone has an event, say `u` is changed after the last step, then `f` needs to be re-evaluated. The FSAL infrastructure should do this (it does this for most methods, I don't know if I've tested the symplectic methods specifically). It might also update this array due to how things are aliased in the initialization though.

Reply:
Pushing this off to https://github.com/SciML/OrdinaryDiffEq.jl/issues/1317 since this PR already does a lot and I don't think this is worth blocking it for.